intro-to-tests/practice-intro-to-pytest.md

```diff
@@ -0,0 +1,910 @@
+# Practice: Intro to Pytest
+
+## Directions
+
+Complete all questions below.
+
+## Practice
+
+### Determining Best Test Cases
+
+<!-- Question 1 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: multiple-choice
+* id: oLlaji
+* title: Intro to Pytest
+##### !question
+
+```python
+def calculate(num_a, num_b, operator):
+    if not (
+        (isinstance(num_a, int) or isinstance(num_a, float)) and
+            (isinstance(num_b, int) or isinstance(num_b, float))):
+        return "Type Error: num_a and num_b must be integer or float"
+    
+    operator = operator.lower()
+    if operator in ["add", "+"]:
+        return num_a + num_b
+    elif operator in ["subtract", "-"]:
+        return num_a - num_b
+    elif operator in ["multiply", "*"]:
+        return num_a * num_b
+    elif operator in ["divide", "/"]:
+        if num_b == 0:
+            return "Zero Division Error"
+        else:
+            return num_a / num_b
+    else:
+        return "Value Error: Operator Not Found"
+```
+
+We want to test `calculate()` and verify that when either `num_a` or `num_b` aren't integers or floats, the function returns `"Type Error: num_a and num_b must be integer or float"`.
+
+Which of these is the best name for a test?
+
+##### !end-question
+##### !options
+
+* `test_calculate()`
+* `test_calculate_int_float_error()`
+* `test_calculate_non_numbers_returns_type_error()`
+* `tests_calculate_non_numbers_returns_type_error()`
+
+##### !end-options
+##### !answer
+
+* `test_calculate_non_numbers_returns_type_error()`
+
+##### !end-answer
+##### !explanation
+
+In tests, long test names are welcome. For each test case, the best test names usually include:
+
+- the name of the tested function
+- context (possibly the kinds of arguments)
+- its expected outcome (usually the return value)
+
+The following guideline is challenging to master, but the best test names tend to describe a situation, and not name exact variable names. If the variable name changes, then the test name could become inaccurate.
+
```

Comment:
Should we include that test names should be in singular form rather than plural? That's the only difference I'm seeing in the last two options.

Reply:
This is actually an artifact of how Pytest and Unittest do testing-- they only recognize that tests start with `test_` or end with `_test`.

Since it's a hard fact (and not a code style/best practice thing), I didn't feel the need to include that, but I'm happy to add that info if it's helpful :O

Reply:
Okay! I trust your judgment on this :)